gulpfile.js

```diff
@@ -6,245 +6,131 @@
-    console.log(cmd);
-    console.log('------------------------------------------------------------');
-    result = execSync(cmd, { encoding: 'utf8' });
-    console.log('Output:');
-    console.log(result);
-  } catch (err) {
-    console.log('Error:');
-    console.log(`An error occurred while generating client for package: "${packageName}":\n ${err.stderr}`);
-  }
-  console.log(`>>>>>>>>>>>>>>>>>>>>>End: "${packageName}" >>>>>>>>>>>>>>>>>>>>>>>>>\n`);
-  return;
-}
+    if (!package || package === packageName || packageName.endsWith(`-${package}`)) {
+      console.log(`>>>>>>>>>>>>>>>>>>> Start: "${packageName}" >>>>>>>>>>>>>>>>>>>>>>>>>`);
 
-function installAutorest() {
-  let installation;
-  let isSuccessful = true;
-  let autorestAlreadyInstalled = true;
-  try {
-    execSync(`autorest --help`);
-  } catch (error) {
-    autorestAlreadyInstalled = false;
-  }
-  try {
-    if (!autorestAlreadyInstalled) {
-      console.log('Looks like autorest is not installed on your machine. Installing autorest . . .');
-      let installCmd = 'npm install -g autorest';
-      console.log(installCmd);
-      installation = execSync(installCmd, { encoding: 'utf8' });
-      //console.debug('installation');
-    }
-    isSuccessful = getAutorestVersion();
-  } catch (err) {
-    isSuccessful = false;
-    console.log(`An error occurred while installing autorest via npm:\n ${util.inspect(err, { depth: null })}.`);
-  }
-  return isSuccessful;
-}
+      const readmeFilePath = path.resolve(path.dirname(nodejsReadmeFilePath), 'readme.md');
 
-function codegen(projectObj, index) {
-  let versionSuccessfullyFound = true;
-  let usingAutoRestVersion = defaultAutoRestVersion;
-  function checkAutorestVersion(actualProj) {
-    if (actualProj.autoRestVersion) {
-      usingAutoRestVersion = actualProj.autoRestVersion;
-    }
-    if (index === 0) {
-      versionSuccessfullyFound = getAutorestVersion(usingAutoRestVersion);
-      if (!versionSuccessfullyFound) {
-        process.exit(1);
+      let cmd = `autorest --nodejs --node-sdks-folder=${azureSDKForNodeRepoRoot} --license-header=MICROSOFT_MIT_NO_VERSION ${readmeFilePath}`;
+      if (use) {
+        cmd += ` --use=${use}`;
       }
-    }
-  }
 
-  function iterateProject(proj, specRoot, usingAutoRestVersion) {
-    for (const key in proj) {
-      if (proj[key]['packageName']) {
-        if (!versionSuccessfullyFound) {
-          checkAutorestVersion(proj[key], index);
-        }
-        generateProject(proj[key], specRoot, usingAutoRestVersion);
-      } else {
-        iterateProject(proj[key], specRoot, usingAutoRestVersion);
+      try {
+        console.log('Executing command:');
+        console.log('------------------------------------------------------------');
+        console.log(cmd);
+        console.log('------------------------------------------------------------');
+        const result = execSync(cmd, { encoding: 'utf8' });
```

Comment:
This PR might not be the time, but I'd really like to drill into why we can't do generation in parallel.

Reply:
I'm sure we can, but you're totally right that this isn't the PR for it.